ui/raidboss/data/06-ew/raid/p5s.ts

```diff
@@ -1,13 +1,210 @@
+  target?: string;
+  topazRays: { [time: number]: (keyof typeof directions | undefined)[] };
+}
+
+const convertAbilityIdToTopazRayIndex = (id: string): number => {
+  // 7703 is the Topaz Ray cast with the lowest cast time
+  return parseInt(id, 16) - parseInt('7703', 16);
+};
+
+const convertCoordinatesToDirection = (x: number, y: number): keyof typeof directions | undefined => {
+  if (x > 100)
+    return y < 100 ? 'NE' : 'SE';
+  if (x < 100)
+    return y < 100 ? 'NW' : 'SW';
+  return undefined;
+};
 
 const triggerSet: TriggerSet<Data> = {
   zoneId: ZoneId.AbyssosTheFifthCircleSavage,
   timelineFile: 'p5s.txt',
-  triggers: [],
+  initData: () => {
+    return {
+      topazRays: {},
+    };
+  },
+  triggers: [
+    {
+      id: 'P5S Attack',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '7A0E', source: 'Proto-Carbuncle' }),
+      run: (data, matches) => data.target = matches.target,
+    },
+    {
+      id: 'P5S Provoke',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '1D6D' }),
+      run: (data, matches) => data.target = matches.source,
+    },
+    {
+      id: 'P5S Sonic Howl',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '7720', source: 'Proto-Carbuncle', capture: false }),
+      response: Responses.aoe(),
+    },
+    {
+      id: 'P5S Venomous Mass',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '771D', source: 'Proto-Carbuncle', capture: false }),
+      alarmText: (data, _matches, output) => {
+        if (data.role === 'tank' && data.target !== data.me)
+          return output.tankSwap!();
+      },
+      alertText: (data, _matches, output) => {
+        if (data.role === 'tank' && data.target !== data.me)
+          return;
+
+        if (data.target === data.me)
+          return output.busterOnYou!();
+        return output.busterOnTarget!({ player: data.ShortName(data.target) });
+      },
+      outputStrings: {
+        busterOnYou: Outputs.tankBusterOnYou,
+        busterOnTarget: Outputs.tankBusterOnPlayer,
+        tankSwap: Outputs.tankSwap,
+      },
+    },
+    {
+      // Maybe too noisy?
+      id: 'P5S Toxic Crunch',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '784A', source: 'Proto-Carbuncle', capture: false }),
+      infoText: (data, _matches, output) => {
+        if (data.role === 'tank' && data.target !== data.me)
+          return;
+
+        if (data.target === data.me)
+          return output.busterOnYou!();
+        return output.busterOnTarget!({ player: data.ShortName(data.target) });
+      },
+      outputStrings: {
+        busterOnYou: Outputs.tankBusterOnYou,
+        busterOnTarget: Outputs.tankBusterOnPlayer,
+      },
+    },
+    {
+      id: 'P5S Double Rush',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '771B', source: 'Proto-Carbuncle', capture: false }),
+      response: Responses.knockback(),
```

Comment:
This is a 5.7s cast, followed by a 2nd one 2.1s later, does this catch both?

Reply:
![image](https://user-images.githubusercontent.com/8849608/187548398-5995b449-5ae4-416f-bd02-e9dec8c21e5f.png)

Just the first one.

Reply:
This should have a delay so it covers both, right?